app/controllers/api/services_controller.rb

```diff
@@ -52,7 +52,11 @@ def create
   end
 
   def update
-    if @service.update_attributes(params[:service])
+    if params[:refresh].presence && @service.discovered?
```

Comment:
The refresh is on update?
Why not making it as totally separate action?

Reply:
To keep it CRUD.